components/edit-func.js

```diff
@@ -2,16 +2,21 @@
 import { PencilSquareIcon } from "@heroicons/react/20/solid";
 import dynamic from "next/dynamic";
 import PropTypes from "prop-types";
-import React from "react";
+import { useContext } from "react";
 // components
 import { Button, ButtonLink } from "./form-elements";
+import SessionContext from "./session-context";
+// lib
+import { itemToSchema } from "../lib/schema";
 /* istanbul ignore file */
 
-export function canEdit(item, actions = ["edit", "edit-json"]) {
-  if ("actions" in item) {
-    return item.actions.find((act) => actions.includes(act.name)) !== undefined;
-  }
-  return false;
+/**
+ * Determines whether the given schema represents an object type that people can edit.
+ * @param {object} schema For type being tested for editability
+ * @returns {boolean} True if the schema represents an object type people can edit
+ */
+export function canEdit(schema) {
+  return Boolean(schema.identifyingProperties?.length > 0);
```

Comment:
Why does `identifyingProperties` determine if an object with some schema can be editable?

Also, I think with this the Add and Edit buttons now appear no matter if a user is logged in and has Adding/Editing permissions on an object.

If I'm not logged in and I go to the schema list, I see the buttons to add. But clicking them just brings me to an empty JSON editor. Similarly, if I'm not logged in and I got to an object page and edit it, the Save buttun is always greyed out (rightfully). But we probably shouldn't even show the buttons for people who don't have Add/Edit permissions or who aren't logged in.

Reply:
> Why does identifyingProperties determine if an object with some schema can be editable?

I don’t know. That was established even before I started in ENCODE.

> If I'm not logged in and I go to the schema list, I see the buttons to add

Dang! That was an obvious one I completely failed to think about. Thanks!